src/main/java/edu/ucsb/cs156/happiercows/controllers/StudentController.java

```diff
@@ -41,6 +44,27 @@ public Student getById(@RequestParam Long id) throws EntityNotFoundException
         return student;
     }
 
+    @Operation(summary= "Update a single student by id")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PutMapping("")
+    public Student updateStudent(
+            @RequestParam Long id,
+            @RequestBody Student incoming) {
+
+        Student student = studentRepository.findById(id).orElseThrow(() -> new EntityNotFoundException(Student.class, id));
+        coursesRepository.findById(incoming.getCourseId()).orElseThrow(() -> new EntityNotFoundException(Courses.class, incoming.getCourseId()));
+
+        student.setCourseId(incoming.getCourseId());
```

Comment:
We would not want to allow changing the course id.  A student that was added to the wrong course should be deleted and added to the correct course.
```suggestion
```